src/coreclr/nativeaot/System.Private.CoreLib/src/System/InvokeUtils.cs

```diff
@@ -128,121 +128,312 @@ private static Exception ConvertOrWidenPrimitivesEnumsAndPointersIfPossible(obje
+                            break;
+                        case EETypeElementType.UInt16:
+                            charValue = (char)Unsafe.As<byte, ushort>(ref rawSrcValue);
+                            break;
+                        default:
+                            goto Failure;
+                    }
+                    rawDstValue = &charValue;
                     break;
 
                 case EETypeElementType.SByte:
-                    sbyte sbyteValue = Convert.ToSByte(srcObject);
-                    dstObject = dstEEType->IsEnum ? Enum.ToObject(dstEEType, sbyteValue) : sbyteValue;
+                    switch (srcElementType)
+                    {
+                        case EETypeElementType.SByte:
+                            sbyteValue = Unsafe.As<byte, sbyte>(ref rawSrcValue);
+                            break;
+                        default:
+                            goto Failure;
+                    }
+                    rawDstValue = &sbyteValue;
                     break;
 
-                case EETypeElementType.Int16:
-                    short shortValue = Convert.ToInt16(srcObject);
-                    dstObject = dstEEType->IsEnum ? Enum.ToObject(dstEEType, shortValue) : shortValue;
+                case EETypeElementType.Byte:
+                    switch (srcElementType)
+                    {
+                        case EETypeElementType.Byte:
+                            byteValue = rawSrcValue;
+                            break;
+                        default:
+                            goto Failure;
+                    }
+                    rawDstValue = &byteValue;
                     break;
 
-                case EETypeElementType.Int32:
-                    int intValue = Convert.ToInt32(srcObject);
-                    dstObject = dstEEType->IsEnum ? Enum.ToObject(dstEEType, intValue) : intValue;
+                case EETypeElementType.Int16:
+                    switch (srcElementType)
+                    {
+                        case EETypeElementType.SByte:
+                            shortValue = (short)Unsafe.As<byte, sbyte>(ref rawSrcValue);
+                            break;
+                        case EETypeElementType.Byte:
+                            shortValue = (short)rawSrcValue;
+                            break;
+                        case EETypeElementType.Int16:
+                            shortValue = Unsafe.As<byte, short>(ref rawSrcValue);
+                            break;
+                        default:
+                            goto Failure;
+                    }
+                    rawDstValue = &shortValue;
                     break;
 
-                case EETypeElementType.Int64:
-                    long longValue = Convert.ToInt64(srcObject);
-                    dstObject = dstEEType->IsEnum ? Enum.ToObject(dstEEType, longValue) : longValue;
+                case EETypeElementType.UInt16:
```

Comment:
```suggestion
                case EETypeElementType.UInt16:
                case EETypeElementType.Char:
```
And delete the Char case

Reply:
Fixed in 62c367487a0ac374602aa59a737f0907eaab3f3f.